Patch by krh@redhat.com:

      In a multihead setup, if only the first screen can be
      initialized, but the second screen is mentioned first in the
      ServerLayout section, the xf86InitOrigins() function will crash
      because the screen referred to in the e.g. "RightOf" part is
      non-existent.

Index: a/hw/xfree86/common/xf86Cursor.c
===================================================================
--- a/hw/xfree86/common/xf86Cursor.c.orig	2007-01-23 07:39:15.000000000 +0200
+++ a/hw/xfree86/common/xf86Cursor.c	2007-02-08 15:45:02.000000000 +0200
@@ -569,6 +569,14 @@
 
 	    screen = &xf86ConfigLayout.screens[i];
 
+	    if (screen->refscreen != NULL &&
+		screen->refscreen->screennum >= xf86NumScreens) {
+		screensLeft &= ~(1 << i);
+	        xf86Msg(X_WARNING, "Not including screen \"%s\" in origins calculation.\n",
+			screen->screen->id);
+	        continue;
+	    }	      
+
 	    switch(screen->where) {
 	    case PosObsolete:
 		OldStyleConfig = TRUE;
